modules/core/include/opencv2/core/types.hpp

```diff
@@ -63,6 +63,161 @@ namespace cv
 //! @addtogroup core_basic
 //! @{
 
+///////////////////////////////////////// Enum operators ///////////////////////////////////////
+
+/**
+
+Provides compatibility operators for both classical and C++11 enum classes,
+as well as exposing the C++11 enum class members for backwards compatibility
+
+@code
+    // Provides operators required for flag enums
+    CV_FLAGS_ENUM(AccessFlag);
+
+    // Provides operators required for flag enum classes
+    CV_FLAGS_ENUM_CLASS(AccessFlag);
+
+    // Exposes AccessFlag::ACCESS_READ to the current namespace
+    CV_ENUM_CLASS_MEMBER(AccessFlag, ACCESS_READ);
+
+    // Exposes the listed 5 members of the enum class AccessFlag to the current namespace
+    CV_ENUM_CLASS_EXPOSE(AccessFlag, 5, ACCESS_READ, ACCESS_WRITE, ACCESS_RW, ACCESS_MASK, ACCESS_FAST);
+@endcode
```

Comment:
> Making all enums enum class would be, however, a big disaster (without any backward compatibility stuff).

I managed to get C++11 `enum class`es working with backwards compatibility and minimal changes.

Basically, instead of:
```C++
enum xyz { m1, m2, m3};
```
one would define:
```C++
enum class xyz { m1, m2, m3};
CV_ENUM_CLASS_EXPOSE(xyz, 3, m1, m2, m3);
```
and in case it is a flag enum, then `CV_FLAGS_ENUM_CLASS(AccessFlag);` instead of `CV_FLAGS_ENUM(AccessFlag);` would do the trick.

Only Python module still have issues with that, but I believe I can work around that as well.
@alalek @hrnr Shall I consider changing the `enum`s into `enum class`es?